src/services/MetricsService/FilteredMetricsService.ts

```diff
@@ -0,0 +1,193 @@
+    return this.serialPromiseQueue.add(() => {
+      switch (eventWithContext.type) {
+        case EventTypeMetric.Installed:
+          return this.publishInstalledEventIfNecessary();
+        case EventTypeMetric.Onboarded:
+          return this.stateStorage.markOnboardedEventShouldBePublished();
+        case EventTypeMetric.Exposed:
+          return this.publishEvent(EventTypeMetric.Exposed, []);
+        case EventTypeMetric.OtkNoDate:
+          if (eventWithContext.exposureHistory.length > 0) {
+            return this.publishEvent(EventTypeMetric.OtkNoDate, []);
+          }
+          break;
+        case EventTypeMetric.OtkWithDate:
+          return this.publishEvent(EventTypeMetric.OtkWithDate, []);
+        case EventTypeMetric.EnToggle:
+          return this.publishEnToggleEventIfNecessary(eventWithContext.state, eventWithContext.onboardedDate);
+        case EventTypeMetric.ExposedClear:
+          if (eventWithContext.exposureStatus.type === ExposureStatusType.Exposed) {
+            return this.publishEvent(EventTypeMetric.ExposedClear, [
+              [
+                'hoursSinceExposureDetectedAt',
+                String(getHoursBetween(getCurrentDate(), new Date(eventWithContext.exposureStatus.exposureDetectedAt))),
+              ],
+            ]);
+          }
+          break;
+      }
+
+      return Promise.resolve();
+    });
+  }
+
+  sendDailyMetrics(systemStatus: SystemStatus, notificationStatus: Status): Promise<void> {
+    return this.serialPromiseQueue.add(() => {
+      return this.publishOnboardedEventIfNecessary(
+        String(notificationStatus),
+        String(systemStatus === SystemStatus.Active),
+      ).then(() => this.metricsService.sendDailyMetrics());
+    });
+  }
+
+  private publishInstalledEventIfNecessary(): Promise<void> {
+    const publishAndMark = (): Promise<void> => {
+      return this.publishEvent(EventTypeMetric.Installed, [], true).then(() =>
+        this.stateStorage.markInstalledEventAsPublished(),
+      );
+    };
+    return this.stateStorage.isInstalledEventPublished().then(isPublished => {
+      if (isPublished === false) {
+        return publishAndMark();
+      }
+    });
+  }
+
+  private publishOnboardedEventIfNecessary(notificationStatus: string, frameworkenabled: string): Promise<void> {
+    const publishAndMark = (): Promise<void> => {
+      return this.publishEvent(EventTypeMetric.Onboarded, [
+        ['pushnotification', notificationStatus],
+        ['frameworkenabled', frameworkenabled],
+      ]).then(() => this.stateStorage.markOnboardedEventShouldNotBePublished());
+    };
+    return this.stateStorage.shouldOnboardedEventBePublished().then(shouldPublish => {
+      if (shouldPublish) {
+        return publishAndMark();
+      }
+    });
+  }
+
+  private publishEnToggleEventIfNecessary(state: boolean, onboardedDate: Date | undefined): Promise<void> {
+    function shouldPublishEnToggle(): boolean {
+      if (state) {
```

Comment:
Should be able to do an early return here and remove the nested if

if(!state) return...

Reply:
Will change